### ISSUE TITLE
Add the ability to rotate CaptureFromWebCamTexture output

### ISSUE THREAD

**Issue by tomkail** [RenderHeads/UnityPlugin-AVProMovieCapture#235]
On iOS, video from webcam texture appears rotated by 90 degrees when held in portrait mode. Since this is never desirable, having the ability to rotate footage would be really handy, especially for recording - the WebCamTexture can be shown on a rotated plane, but the output footage that's saved to disk should really be shown in a rotation that the user expects.

I imagine you could use a shader to rotate the footage into a new texture, rotated WebCamTexture, and then pass that to CaptureFromWebCamTexture.
Alternatively, use FFMpeg to rotate the footage when it's saved to a file.

Before I try either though, I wanted to know if there's a best practice approach for this. Would these ideas work? Are they advisable? 

Thanks!

**Comment by tomkail** [comment]
Is anyone able to comment on the best way to approach this?

**Comment by tomkail** [comment]
Anybody? We'd really like to know if there's an advisable way to do this before we hack around it.

**Comment by MorrisRH** [comment]
I think the best approach would be to use the CaptureFromCamera component:
- Set the camera resolution to match that of the webcamtexture
- Render the webcamtexture to a full screen quad or using uGUI
- Orient the camera correctly for the current device orientation

Should we add support for this it would be by adding support for writing out the transform metadata to the captured video.

**Comment by MorrisRH** [comment]
Linking to #234

**Comment by briveramelo** [comment]
I'm also interested in this. Use case is taking selfie videos. 

Right now I'm using the hack tomkail suggested by rotating with a shader, but watching these videos in any other context besides the app is still impractical.

In the meantime for anyone else interested, here's the shader. Basic idea is to create a material with this shader and attach it to a RawImage UI element. 
```
Shader "Unlit/Rotator"
{
    Properties
    {
        _MainTex ("Texture", 2D) = "white" {}
    }
    SubShader
    {
        Tags { "RenderType"="Opaque" }
        LOD 100

        Pass
        {
            CGPROGRAM
            #pragma vertex vert
            #pragma fragment frag

            #include "UnityCG.cginc"

            struct appdata
            {
                float4 vertex : POSITION;
                float2 uv : TEXCOORD0;
            };

            struct v2f
            {
                float2 uv : TEXCOORD0;
                UNITY_FOG_COORDS(1)
                float4 vertex : SV_POSITION;
            };

            sampler2D _MainTex;
            float4 _MainTex_ST;
            float _Rotation;

            float2 RotateMat(float2 uv, float t)
            {
                float2 newUV;
                newUV = uv * 2 - 1;
                float c = cos(_Rotation + t);
                float s = sin(_Rotation + t);

                float2x2 mat = float2x2(c, -s, s, c);
                newUV = mul(mat, newUV);
                newUV = newUV * 0.5 + 0.5;
                return newUV;
            }

            v2f vert (appdata v)
            {
                v2f o;
                o.vertex = UnityObjectToClipPos(v.vertex);
                o.uv.xy = RotateMat(v.uv.xy, 1.57);
                return o;
            }

            fixed4 frag (v2f i) : SV_Target
            {
                // sample the texture
                fixed4 col = tex2D(_MainTex, i.uv);
                return col;
            }
            ENDCG
        }
    }
}

```

**Comment by Chris-RH** [comment]
Fixed in AVPro Movie Capture 5.2.0
